nncf/experimental/onnx/datasets/imagenet_dataset.py

```diff
@@ -47,18 +47,30 @@ def __len__(self) -> int:
         return len(self.dataset)
 
 
-def infer_input_shape(model: ModelProto) -> Tuple[Optional[int], int, int, int]:
-    assert len(model.graph.input) == 1
-
+def infer_input_shape(model: ModelProto, main_shape: List[int] = None) -> Tuple[List[int], List[str]]:
+    input_shape = None
+    input_keys = None
     for _input in model.graph.input:
-        dim = _input.type.tensor_type.shape.dim
-        input_shape = [MessageToDict(d).get("dimValue") for d in dim]
-
-    input_shape = [int(dim) if dim is not None else dim for dim in input_shape]
-
-    assert len(input_shape) == 4
-
-    return input_shape
+        if len(_input.type.tensor_type.shape.dim) == 4:
+            dim = _input.type.tensor_type.shape.dim
+            _input_shape = [int(MessageToDict(d).get("dimValue")) for d in dim]
+            if main_shape is not None:
+                if main_shape == _input_shape:
+                    input_shape = main_shape
+                    input_keys = _input.name
+                    break
+            else:
+                if input_shape is None and input_keys is None:
+                    input_shape = _input_shape
+                    input_keys = _input.name
+                else:
+                    if input_shape[2] < _input_shape[2]:
```

Comment:
Why does it work like that? Could you add some comments in the code?

Reply:
Done